Zemismart/__init__.py

```diff
@@ -111,17 +157,47 @@ def handleNotification(self, handle, data):
             elif data[1] == self.get_position_cmd[0]:
                 pos = data[5]
                 self.save_position(pos)
-                self.last_command_status = True
-            elif data[1] in (self.finished_moving_cmd[0], self.notify_position_cmd[0]):
+                self._get_position_responses.add(data[1])
+                self._check_get_position_response()
+            elif data[1] == self.finished_moving_cmd[0]:
                 pos = data[4]
                 self.save_position(pos)
                 self.last_command_status = True
-            elif data[1] in (self.set_position_cmd[0], self.pin_cmd[0], self.move_cmd[0]):
+            elif data[1] == self.unknown_cmd_a9[0]:
+                self._get_position_responses.add(data[1])
+                self._check_get_position_response()
+            elif data[1] == self.get_timers_cmd[0]:
+                self._get_position_responses.add(data[1])
+                self.timers = []
+                # Single timer takes 5 bytes
+                for offset in range(0, int(int(data[2]) / 5)):
+                    timer_data_start = 3 + offset * 5
+                    timer = Zemismart.Timer(
+                        bool(data[timer_data_start]),
+                        int(data[timer_data_start + 1]),
+                        int(data[timer_data_start + 2]),
+                        int(data[timer_data_start + 3]),
+                        int(data[timer_data_start + 4])
+                    )
+                    self.timers.append(timer)
+
+                self._check_get_position_response()
+            elif data[1] in (self.set_position_cmd[0], self.pin_cmd[0], self.move_cmd[0], self.update_timer_cmd[0]):
                 if data[3] == 0x5A:
                     self.last_command_status = True
                 elif data[3] == 0xA5:
                     self.last_command_status = False
 
+    def _check_get_position_response(self):
+        """
+        Command 0xA7 (get_position) yields three responses from the device. We can't treat the command result as
+        a success, unless we have receieved all of them.
+        This method checks that since the last update() call, where self._get_position_responses is reset, we
+        receieved all the required responses.
+        """
+        if self._get_position_responses == {self.get_position_cmd[0], self.get_timers_cmd[0], self.unknown_cmd_a9[0]}:

```

Comment:
Additional commands, other than the used here, could be added to the set (maybe by mistake), so I would prefer to use `issuperset()`
```suggestion
        if self._get_position_responses.issuperset({self.get_position_cmd[0], self.get_timers_cmd[0], self.unknown_cmd_a9[0]}):
```
or `>=`
```suggestion
        if self._get_position_responses >= {self.get_position_cmd[0], self.get_timers_cmd[0], self.unknown_cmd_a9[0]}:
```

But I think `issuperset()` is best because it is more clear what it does.